source/create_coupling_sparsity_pattern_with_exact_intersections.cc

```diff
@@ -108,8 +115,16 @@ namespace dealii::NonMatching
           }
       }
 
+    const bool dof_mask_is_active =
+      dof_mask.n_rows() == n_space_dofs && dof_mask.n_cols() == n_immersed_dofs;
+    Assert(
```

Comment:
I had to add this Assert for the case when `dof_mask` is active and I have non-trivial masks, as it seems to me that it's currently not implemented ( seehttps://github.com/dealii/dealii/blob/c487e7fe81417c23c3d8d4dff437665f9a827916/include/deal.II/lac/affine_constraints.templates.h#L4406)